doc/README.md

```diff
@@ -0,0 +1,39 @@
+# Documentation
+
+This folder contains detailed documentation of the library's functionality. Its target audience are sophisticated users of the library, as well as developers that wish to contribute to / extend it.
+
+## Table of contents
+
+- [Lifecycling](./Lifecycling.md)
+- Valuation semantics (upcoming)
+- Utility functions (upcoming)
+
+## Prerequisite knowledge
+
+It is important that the reader is familiar with the papers [[1]](#1) and [[2]](#2), which introduce the model for a financial instrument that the library implements. Specifically, the primitives from [[2]](#2) are used in the definition of the `Claim` type.
+
+The section on the contracts' valuation semantics uses results and notation from measure-theoretic probability theory and the theory of no-arbitrage pricing.
+
+Finally, those who wish to develop the library would benefit from an understanding of recursion schemes, which are used to implement most of the workflows. Patrick Thomson's [blog post series](https://blog.sumtypeofway.com/posts/introduction-to-recursion-schemes.html) is an excellent resource to get up to speed on the topic.
```

Comment:
I don't know if it's worth having this in a separate document. I think it's difficult to address two audiences: users vs developers, in the same doc.

You don't really need to know anything about recursion schemes to use the library, right? And I think exposing that may create an artifical barrier to entry or impression of one for new users.

Reply:
I agree with you, I will remove this `README` file for now.